### PR TITLE
Feature/pn 2256

### PR DESCRIPTION
**Pull request by masciale89**

### CODE REVIEW

File: src/main/java/it/pagopa/pn/deliverypush/service/impl/SafeStorageServiceImpl.java [hunk 1]
```diff
@@ -29,10 +39,10 @@ public SafeStorageServiceImpl(PnSafeStorageClient safeStorageClient) {
     public FileDownloadResponseInt getFile(String fileKey, Boolean metadataOnly) {
         try {
             FileDownloadResponse fileDownloadResponse = safeStorageClient.getFile(fileKey, metadataOnly);
-
-            return getFileDownloadResponseInt(fileDownloadResponse);
-        } catch (Exception e) {
-            throw new PnInternalException("Cannot getfileinfo", ERROR_CODE_DELIVERYPUSH_GETFILEERROR, e);
+            return getFileDownloadResponseInt( fileDownloadResponse );
+        } catch ( PnInternalException ex ) {
```
Comment (marco-vit-pagopa) [inline src/main/java/it/pagopa/pn/deliverypush/service/impl/SafeStorageServiceImpl.java:43]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

File: src/main/java/it/pagopa/pn/deliverypush/action/utils/AttachmentUtils.java [hunk 2]
```diff
@@ -83,19 +82,20 @@ private void forEachAttachment(NotificationInt notification, Consumer<Notificati
 
     private void checkAttachment(NotificationDocumentInt attachment) {
         NotificationDocumentInt.Ref ref = attachment.getRef();
-
-        FileDownloadResponseInt fd = safeStorageService.getFile(ref.getKey(),true);
+        FileDownloadResponseInt fd = null;
+        try {
+            fd = safeStorageService.getFile(ref.getKey(),true);
+        } catch ( PnNotFoundException ex ) {
```
Comment (marco-vit-pagopa) [inline src/main/java/it/pagopa/pn/deliverypush/action/utils/AttachmentUtils.java:88]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

File: src/main/java/it/pagopa/pn/deliverypush/action/StartWorkflowHandler.java [hunk 3]
```diff
@@ -91,8 +90,7 @@ private void scheduleStartRecipientWorkflow(String iun, Integer recIndex) {
     }
     
     private void handleValidationError(NotificationInt notification, PnValidationException ex) {
-        List<String> errors =  ex.getProblem().getErrors().stream()
-                .map(ProblemError::getDetail).collect(Collectors.toList());
+        List<String> errors = Collections.singletonList( ex.getProblem().getDetail() );
         log.info("Notification refused, errors {} - iun {}", errors, notification.getIun());
         addTimelineElement( timelineUtils.buildRefusedRequestTimelineElement(notification, errors), notification);
```
Comment (simonerondinella) [inline src/main/java/it/pagopa/pn/deliverypush/action/StartWorkflowHandler.java:95]: Problem qui potrebbe essere null? Fare un check di problem diverso da null, se problem è null la lista di errors sarà vuota

**Review by simonerondinella — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pn/deliverypush/action/StartWorkflowHandler.java (hunk 3 above)
Comment (masciale89) [reply src/main/java/it/pagopa/pn/deliverypush/action/StartWorkflowHandler.java]: non dovrebbe essere null... aggiunto comunque il controllo per evitare NPE

**Review by simonerondinella — APPROVED** [review]

**Pull request merged**